src/Windows/Avalonia.Win32/TrayIconImpl.cs

```diff
@@ -0,0 +1,295 @@
+            {
+                SystemDecorations = SystemDecorations.None,
+                SizeToContent = SizeToContent.WidthAndHeight,
+                Background = null,
+                TransparencyLevelHint = WindowTransparencyLevel.Transparent,
+                Content = new TrayIconMenuFlyoutPresenter()
+                {
+                    Items = _exporter.GetMenu()
+                }
+            };
+
+            GetCursorPos(out POINT pt);
+
+            _trayMenu.Position = new PixelPoint(pt.X, pt.Y);
+
+            _trayMenu.Show();
+        }
+
+        /// <summary>
+        /// Custom Win32 window messages for the NotifyIcon
+        /// </summary>
+        enum CustomWindowsMessage : uint
+        {
+            WM_TRAYICON = WindowsMessage.WM_APP + 1024,
+            WM_TRAYMOUSE = WindowsMessage.WM_USER + 1024
+        }
+
+        class TrayIconMenuFlyoutPresenter : MenuFlyoutPresenter, IStyleable
+        {
+            Type IStyleable.StyleKey => typeof(MenuFlyoutPresenter);
+
+            public override void Close()
+            {
+                // DefaultMenuInteractionHandler calls this
+                var host = this.FindLogicalAncestorOfType<TrayPopupRoot>();
+                if (host != null)
+                {
+                    SelectedIndex = -1;
+                    host.Close();
+                }
+            }
+        }
+
+        class TrayPopupRoot : Window
+        {
+            private ManagedPopupPositioner _positioner;
+
+            public TrayPopupRoot()
+            {
+                _positioner = new ManagedPopupPositioner(new TrayIconManagedPopupPositionerPopupImplHelper(MoveResize));
+                Topmost = true;
+
+                Activated += TrayPopupRoot_Activated;
+                Deactivated += TrayPopupRoot_Deactivated;
+
+                LostFocus += TrayPopupRoot_LostFocus1;
+
+                ShowInTaskbar = false;
+            }
+
+            private void TrayPopupRoot_LostFocus1(object sender, Interactivity.RoutedEventArgs e)
+            {
+                Debug.WriteLine("TrayIcon - Lost Focus");
+            }
+
+            private void TrayPopupRoot_Activated(object sender, EventArgs e)
+            {
+                Debug.WriteLine("TrayIcon - Activated");
+            }
+
+            private void TrayPopupRoot_Deactivated(object sender, EventArgs e)
+            {
+                Debug.WriteLine("TrayIcon - Deactivated");
+
+                Dispatcher.UIThread.Post(() =>
+                {
+                    Close();
+                });
+            }
+
+            private void MoveResize(PixelPoint position, Size size, double scaling)
+            {
+                PlatformImpl.Move(position);
+                PlatformImpl.Resize(size, PlatformResizeReason.Layout);
+            }
+
+            private void TrayPopupRoot_LostFocus(object sender, Interactivity.RoutedEventArgs e)
+            {
+                Close();
+            }
```

Comment:
Private method without any references, do we need it?

Reply:
I think we do actually,this code broken behaviour of the trayicon popup I think.

Reply:
(partially,something is wrong with focus also)